Kernel/Arch/aarch64/Exceptions.cpp

```diff
@@ -78,6 +78,13 @@ static void setup_el1()
 
     Aarch64::SCTLR_EL1::write(system_control_register_el1);
 
+    Aarch64::CPACR_EL1 cpacr_el1 = {};
+    cpacr_el1.ZEN = 0;     // Trap SVE instructions at EL1 and EL0
+    cpacr_el1.FPEN = 0b11; // Don't trap Advanced SIMD and floating-point instructions
```

Comment:
The kernel shouldn't use any floating point instructions (…else we'd have to save the float registers on context switch), right? Why do we need this?

Reply:
We need this because userspace (EL0) will execute floating-point instructions. According to the docs (https://developer.arm.com/documentation/ddi0595/2021-03/AArch64-Registers/CPACR-EL1--Architectural-Feature-Access-Control-Register?lang=en#fieldset_0-21_20) there is no option to only trap on floating-point instruction in EL1, it's only possible to completely disable the trapping for both EL0 and EL1.

I went ahead and implemented storing/restoring of the floating point registers for the userspace threads, because we need that anyway.